css-highlight-api-1/Overview.bs

```diff
@@ -225,41 +214,45 @@ Registering Custom Highlights</h3>
 
 	[Exposed=Window]
 	interface HighlightRegistry {
-		setlike<Highlight>;
+		maplike<DOMString, Highlight>;
 	};
 	</xmp>
 
 	<div algorithm="to register a custom highlight">
 		To [=register=] a [=custom highlight=],
-		invoke the {{HighlightRegistry/add()}} method of the [=highlight registry=]
-		with the [=custom highlight=] as the argument.
-
-		When invoked,
-		the <dfn method for=HighlightRegistry>add(Highlight value)</dfn> method must run these steps:
-
-		1. If there is already a [=set entry=] with the same {{Highlight/name}} as the {{Highlight/name}} of {{value}},
-			then [=throw=] an {{"OperationError"}}.
-		3. Let |valueArg| be the result of [=converted to an ECMAScript value|converting=] {{value}} to an ECMAScript value.
-		4. Let |result| be the result of running [[webidl#es-add-delete|the steps for a built-in setlike add function]],
-			with the [=context object=] as the <code>this</code> value
-			and with |valueArg| as the argument.
-		5. Return |result|.
-	</div>
+		invoke the <code>set</code> method of the [=highlight registry=]
+		which will run [[webidl#es-map-set|the steps for a built-in maplike set function]],
+		with the [=context object=] as the <code>this</code> value,
+		the passed-in [=custom highlight name=] as |keyArg|,
+		and the passed-in highlight as |valueArg|.
+
+		The <dfn>custom highlight name</dfn> assigned to a [=custom highlight=] when it is [=registered=]
+		is used to identify the highlight during styling (see [[#styling-highlights]]).
+
+	Note: It is possible to [=register=] a [=custom highlight=] with any [=custom highlight name=]
+	that is a valid [[webidl#idl-DOMString|DOMString]], but the name must also be a valid CSS
+	<<ident-token>> in order to be used to style the highlight.
+
+	Note: It is possible to [=register=] a [=custom highlight=] with more than one [=custom highlight name=].
+	However, using more than one name can lead to the highlight being assigned multiple different sets of styles,
+	without a way to control the stacking order of these styles during [[#painting|painting]].
+	This may be limiting for authors and may cause confusing painting behavior.
+	Therefore, <b>it is recommended that authors only use one name per highlight</b>.
```

Comment:
I'm not sure if we should be recommending against the practice altogether, or just recommending against using **both** names to style the highlight, which is what actually causes trouble. If only one of the names is ever used in the stylesheet, there is no confusion. As I discussed in the thread, I can see valid use cases for registering multiple highlights with the same name by library authors to avoid manually keeping the objects in sync, with the expectation of their authors only actually styling one of these names.
I also think it would be good to include your minimized example code where the cascade doesn't work as expected, as that was quite illuminating in understanding the problem.

Reply:
Okay, that works for me. I will update this to reflect "style with one name" and add the example.

Reply:
I have updated this. PTAL.